### PR TITLE
disttask: fix hang in make ut

### LINKED ISSUE #49405: Many test case in `disttask/framework` pkg will timeout when `GOMAXPROCS = 1`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

```
diff --git a/pkg/disttask/framework/framework_dynamic_dispatch_test.go b/pkg/disttask/framework/framework_dynamic_dispatch_test.go
index 2289202158..77ce9013f4 100644
--- a/pkg/disttask/framework/framework_dynamic_dispatch_test.go
+++ b/pkg/disttask/framework/framework_dynamic_dispatch_test.go
@@ -15,6 +15,7 @@
 package framework_test

 import (
+       "runtime"
        "testing"

        "github.com/pingcap/failpoint"
@@ -32,6 +33,10 @@ func TestFrameworkDynamicBasic(t *testing.T) {
 }

 func TestFrameworkDynamicHA(t *testing.T) {
+       val := runtime.GOMAXPROCS(1)
+       defer func() {
+               runtime.GOMAXPROCS(val)
+       }()
        ctx, ctrl, testContext, distContext := testutil.InitTestContext(t, 3)
        defer ctrl.Finish()
```

Apply the diff above, then run `TestFrameworkDynamicHA` test case.

### 2. What did you expect to see? (Required)

test pass

### 3. What did you see instead (Required)

test timeout

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

```
mysql> select tidb_version();
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                 |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v7.6.0-alpha-423-g0b39f99e38-dirty
Edition: Community
Git Commit Hash: 0b39f99e389d6b04b8c2d515d95979535da1a18f
Git Branch: master
UTC Build Time: 2023-12-12 08:08:56
GoVersion: go1.21.5
Race Enabled: false
Check Table Before Drop: false
Store: unistore |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: close #49405 

Problem Summary:

### What changed and how does it work?
set tasks in ut with 1 concurrency
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: pkg/disttask/framework/testutil/disttest_util.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func RegisterRollbackTaskMeta(t *testing.T, ctrl *gomock.Controller, mockDispatc
 func DispatchTask(ctx context.Context, t *testing.T, taskKey string) *proto.Task {
 	mgr, err := storage.GetTaskManager()
 	require.NoError(t, err)
-	_, err = mgr.CreateTask(ctx, taskKey, proto.TaskTypeExample, 8, nil)
+	_, err = mgr.CreateTask(ctx, taskKey, proto.TaskTypeExample, 1, nil)
```
Comment (D3Hunter) [inline pkg/disttask/framework/testutil/disttest_util.go:122]: maybe add this failpoint to all those case
```
require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/util/cpu/mockNumCpu", `return(1)`))
```

File: pkg/disttask/framework/testutil/context.go [hunk 2]
```diff
@@ -41,6 +43,10 @@ func InitTestContext(t *testing.T, nodeNum int) (context.Context, *gomock.Contro
 	defer ctrl.Finish()
 	ctx := context.Background()
 	ctx = util.WithInternalSourceType(ctx, "scheduler")
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/util/cpu/mockNumCpu", "return(1)"))
```
Comment (D3Hunter) [inline pkg/disttask/framework/testutil/context.go:46]:
```suggestion
	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/util/cpu/mockNumCpu", "return(8)"))
```
i remember there're tasks of currency = 4

File: pkg/disttask/framework/testutil/disttest_util.go
Comment (D3Hunter) [inline pkg/disttask/framework/testutil/disttest_util.go:1]: this part can be reverted

**Pull request merged**